server/eclipse-project/src/test/java/au/org/democracydevelopers/corla/csv/DominionCVRExportParserTests.java

```diff
@@ -611,4 +619,167 @@ public void parseWriteIns() throws IOException {
       assertTrue(contest.choices().get(2).qualifiedWriteIn());
     }
   }
+
+  /**
+   * Simple test of successful parsing of a tiny IRV test example, with IRV, STV, and plurality,
+   * in that order.
+   * Tests that all the metadata and some of the votes are correct.
+   * In particular, the STV contest should be completely dropped.
+   * @throws IOException never.
+   */
+  @Test
+  @Transactional
+  public void parseThreeCandidatesTenVotesPlusSTVPlusPluralitySucceedsAndDropsSTV() throws IOException {
+    testUtils.log(LOGGER, "parseThreeCandidatesTenVotesPlusSTVPlusPluralitySucceedsAndDropsSTV");
+    final Path path = Paths.get(TINY_CSV_PATH + "ThreeCandidatesTenVotesPlusSTVPlusPlurality.csv");
+
+    doIRVAndSTVAndPluralityTest(path, "Cheyenne");
+  }
+
+  /**
+   * Second test for successful parsing of mixed IRV, plurality, STV contests. The STV contest is
+   * at the end.
+   * @throws IOException never.
+   */
+  @Test
+  @Transactional
+  public void parseThreeCandidatesTenVotesPlusPluralityPlusSTVSucceedsAndDropsSTV() throws IOException {
+    testUtils.log(LOGGER, "parseThreeCandidatesTenVotesPlusPluralityPlusSTVSucceedsAndDropsSTV");
+    final Path path = Paths.get(TINY_CSV_PATH + "ThreeCandidatesTenVotesPlusPluralityPlusSTV.csv");
+
+    doIRVAndSTVAndPluralityTest(path, "Yuma");
+  }
+
+  /**
+   * Third test for successful parsing of mixed IRV, plurality, STV contests. The STV contest is
+   * at the beginning.
+   * @throws IOException never.
+   */
+  @Test
+  @Transactional
+  public void parseSTVPlusThreeCandidatesTenVotesPlusPluralitySucceedsAndDropsSTV() throws IOException {
+    testUtils.log(LOGGER, "parseSTVPlusThreeCandidatesTenVotesPlusPluralitySucceedsAndDropsSTV");
+    final Path path = Paths.get(TINY_CSV_PATH + "STVPlusThreeCandidatesTenVotesPlusPlurality.csv");
+
+    doIRVAndSTVAndPluralityTest(path, "La Plata");
+  }
+
```

Comment:
Rogue extra line at top of method comment for noContestUploadIsAnError(). Also, some remaining tests have no public, just void.